### PR TITLE
Tournaments/public

### PR DESCRIPTION
**Pull request by haileyavery**
draft pr

### CODE REVIEW

File: Backend/.gitignore
Comment (ZhiLinR) [inline Backend/.gitignore:1]: WHY IS THIS GONE 😭  OMG

File: Backend/Tournament_Public/.env
Comment (ZhiLinR) [inline Backend/Tournament_Public/.env:1]: DUN PUSH ENV FILE 😭  PLS

File: Backend/Tournament_Public/controllers/publicUserController.js
Comment (ZhiLinR) [inline Backend/Tournament_Public/controllers/publicUserController.js:39]: doesnt use the middleware defined in `Backend/Tournament_Public/middlewares/errorHandler.js` ?

File: Backend/Tournament_Public/middlewares/errorHandler.js
Comment (ZhiLinR) [inline Backend/Tournament_Public/middlewares/errorHandler.js:23]: does not follow #19

File: Backend/Tournament_Public/models/tournament.js
Comment (ZhiLinR) [inline Backend/Tournament_Public/models/tournament.js:13]: single quotes

File: Backend/Tournament_Public/models/tournament.js
Comment (ZhiLinR) [inline Backend/Tournament_Public/models/tournament.js:34]: same as above

File: Backend/Tournament_Public/tests/publicUserRoutes.test.js
Comment (ZhiLinR) [inline Backend/Tournament_Public/tests/publicUserRoutes.test.js:53]: remove unused? why commented out

File: Backend/Tournament_Public/.env
Comment (haileyavery) [reply Backend/Tournament_Public/.env:1]: T_T

File: Backend/Tournament_Public/middlewares/errorHandler.js
Comment (haileyavery) [reply Backend/Tournament_Public/middlewares/errorHandler.js:23]: donez pls check

File: Backend/Tournament_Public/models/tournament.js
Comment (haileyavery) [reply Backend/Tournament_Public/models/tournament.js:13]: done

File: Backend/Tournament_Public/routes/publicUserRoutes.js
Comment (ZhiLinR) [inline Backend/Tournament_Public/routes/publicUserRoutes.js:17]: /view/tournaments/upcoming

File: Backend/Tournament_Public/routes/publicUserRoutes.js
Comment (ZhiLinR) [inline Backend/Tournament_Public/routes/publicUserRoutes.js:24]: /view/tournaments/inprogress

File: Backend/Tournament_Public/routes/publicUserRoutes.js
Comment (ZhiLinR) [inline Backend/Tournament_Public/routes/publicUserRoutes.js:31]: /view/tournaments/inprogress

File: Backend/Tournament_Public/routes/publicUserRoutes.js
Comment (ZhiLinR) [inline Backend/Tournament_Public/routes/publicUserRoutes.js:24]: /view/tournaments/events

File: Backend/Tournament_Public/controllers/publicUserController.js
Comment (ZhiLinR) [inline Backend/Tournament_Public/controllers/publicUserController.js:80]: middleware usage? for both success and error responses

File: Backend/Tournament_Public/controllers/publicUserController.js
Comment (ZhiLinR) [inline Backend/Tournament_Public/controllers/publicUserController.js:147]: doesn't use middleware unlike the previous ones

File: Backend/Tournament_Public/package.json
Comment (ZhiLinR) [inline Backend/Tournament_Public/package.json:2]: change to tournament-public

File: Backend/Tournament_Public/userService/userService.js
Comment (ZhiLinR) [inline Backend/Tournament_Public/userService/userService.js:1]: intermediate layer to note, possibly not neccessary?

File: Backend/Tournament_User/TournamentUserMVSC/models/tournamentModel.js
Comment (ZhiLinR) [inline Backend/Tournament_User/TournamentUserMVSC/models/tournamentModel.js:1]: not sure but did u use something from jaspers code cos this got merged into your branch?

**Review by ZhiLinR — COMMENTED** [review]
minor fixes

File: Backend/Tournament_User/TournamentUserMVSC/models/tournamentModel.js
Comment (haileyavery) [reply Backend/Tournament_User/TournamentUserMVSC/models/tournamentModel.js:1]: yeah it was merged with jasper's branch

File: Backend/Tournament_Public/server.js
Comment (warkionw) [inline Backend/Tournament_Public/server.js:4]: Think might need to add CORS for frontend calling here

File: Backend/Tournament_Public/controllers/publicUserController.js
Comment (warkionw) [inline Backend/Tournament_Public/controllers/publicUserController.js:80]: Not sure if i seeing wrong? but i think you only middleware for error and not success.

File: Backend/Tournament_Public/userService/userService.js
Comment (warkionw) [inline Backend/Tournament_Public/userService/userService.js:45]: might wanna add some validation check for tournamentId input eg. If tournamentId is null or empty. If dh can straight up error

Comment (warkionw) [reply Backend/Tournament_Public/userService/userService.js:45]: ngl not sure where the validation shld be i put this comment here cause svc and thats where i put mine in spring boot but idk nodejs so yeah

File: Backend/Tournament_Public/middlewares/errorHandler.js
Comment (warkionw) [inline Backend/Tournament_Public/middlewares/errorHandler.js:16]: not sure if this shld be 400 Bad Request? cause if missing request entry then the proc will just return null. 45000 highly likely is 500 Internal Server Error instead

File: Backend/Tournament_Public/models/config/db.js
Comment (warkionw) [inline Backend/Tournament_Public/models/config/db.js:1]: might wanna move your config folder to main folder (Tournament_Public) instead of inside models ah

File: Backend/Tournament_Public/tests/tournamentController.test.js
Comment (warkionw) [inline Backend/Tournament_Public/tests/tournamentController.test.js:1]: Can add some comments or docs for your test file to state what they are testing

File: Backend/Tournament_Public/tests/tournamentModel.test.js
Comment (warkionw) [inline Backend/Tournament_Public/tests/tournamentModel.test.js:1]: Same with earlier abt the comments

File: Backend/Tournament_User/README.md
Comment (warkionw) [inline Backend/Tournament_User/README.md:1]: hi rmb do the readme ya. can ref to arshiya's or mine for the sample.

File: Backend/Tournament_Public/middlewares/errorHandler.js
Comment (warkionw) [reply Backend/Tournament_Public/middlewares/errorHandler.js:16]: Regarding middleware think can ref jasper's ah he got gc1 approval also

File: Backend/Tournament_Public/controllers/publicUserController.js
Comment (warkionw) [inline Backend/Tournament_Public/controllers/publicUserController.js:18]: Think can maybe change the comment to 'No tournaments found' instead or smt I'm assuming only 0 if got no tournaments ah

File: Backend/Tournament_Public/controllers/publicUserController.js
Comment (warkionw) [inline Backend/Tournament_Public/controllers/publicUserController.js:76]: can change to 'No ongoing tournaments' instead

File: Backend/Tournament_Public/controllers/publicUserController.js
Comment (warkionw) [inline Backend/Tournament_Public/controllers/publicUserController.js:87]: same abt the ongoing

File: Backend/Tournament_Public/controllers/publicUserController.js
Comment (warkionw) [inline Backend/Tournament_Public/controllers/publicUserController.js:1]: need to add a /health end point also, response format can refer to mine/arshiya's README

File: Backend/Tournament_Public/server.js
Comment (warkionw) [reply Backend/Tournament_Public/server.js:4]: When you do this try make the ORIGIN an env var

**Review by warkionw — CHANGES_REQUESTED** [review]
hi rmb to dockerise your container also tq

File: Backend/Tournament_Public/server.js
Comment (haileyavery) [reply Backend/Tournament_Public/server.js:4]: ok  boss. my CORS_ORIGIN=http://localhost:3000 is liddis right in env file

Comment (warkionw) [reply Backend/Tournament_Public/server.js:4]: shld be the frontend addr ah so the port depends on the frontend but can change ltr anyways so not a big deal

File: Backend/Tournament_Public/README.md
Comment (warkionw) [inline Backend/Tournament_Public/README.md:1]: tq looks gd \('0')/

File: Backend/Tournament_Public/controllers/publicUserController.js
Comment (warkionw) [inline Backend/Tournament_Public/controllers/publicUserController.js:122]: nice validation :>

**Review by warkionw — COMMENTED** [review]
lgtm ah i'll let gc1 have final approval ._.

**Review by ZhiLinR — APPROVED** [review]
kool beans

**Pull request merged**
